project/ScroogeRunner.scala

```diff
@@ -0,0 +1,192 @@
+import sbt._
+import Keys._
+import java.io.File
+
+// Runs the freshly-built Scrooge binary (used for integration tests and benchmarking)
+// How to run: `sbt genTestThrift` or `sbt genBenchmarkThrift`
+object ScroogeRunner {
+  sealed abstract class Language(val scroogeName: String)
+  case object Scala extends Language("scala")
+  case object ApacheJava extends Language("java")
+  case object Java extends Language("experimental-java")
+
+  val javaNamespace =
+    """-n thrift.defaults=thrift.java_defaults
+      |-n thrift.colors=thrift.java_colors
+      |-n thrift.test=thrift.java_test
+      |-n thrift.test1=thrift.java_test1
+      |-n thrift.test2=thrift.java_test2
+      |-n thrift.collision=thrift.java_collision
+      |-n thrift.constants=thrift.java_constants
+      |-n thrift.def.default=thrift.java_def.default
+      |-n foo=java_foo
+      |-n bar=java_bar
+      |-n com.fake=com.java_fake
+      |-n com.twitter.scrooge.integration_scala=com.twitter.scrooge.integration_java
+      |--default-java-namespace java_thrift
+    """.stripMargin
+
+  val apacheJavaNamespace =
+    """-n thrift.defaults=thrift.apache_java_defaults
+      |-n thrift.colors=thrift.apache_java_colors
+      |-n thrift.test=thrift.apache_java_test
+      |-n thrift.test1=thrift.apache_java_test1
+      |-n thrift.test2=thrift.apache_java_test2
+      |-n thrift.collision=thrift.apache_java_collision
+      |-n thrift.def.default=thrift.apache_java_def.default_ns
+      |-n foo=apache_java_foo
+      |-n bar=apache_java_bar
+      |-n com.fake=com.apache_java_fake
+      |-n com.twitter.scrooge.integration_scala=com.twitter.scrooge.integration_apache_java
+      |--default-java-namespace apache_java_thrift
+    """.stripMargin
+
+  val namespaces = Map[Language, String](
+    Java -> javaNamespace,
+    ApacheJava -> apacheJavaNamespace,
+    Scala -> ""
+  )
+
+  type Input[T] = Types.Id[T]
+
+  class Runner(out: Input[Keys.TaskStreams], base: Input[File], cp: Input[Classpath], outputDir: Input[File]) {
+    def run(language: Language, namespace: String, finagle: Boolean = true, args: String) : Unit =  {
+      val finagleArg = if(finagle) "--finagle" else ""
+
+      val command = "java -cp " +
```

Comment:
Scrooge is using scala 2.10 now, so feel free to use string interpolation in the future.